export/export.go

```diff
@@ -59,28 +83,78 @@ func ExportBranchStart(paradeDB parade.Parade, cataloger catalog.Cataloger, repo
 	return exportID, err
 }
 
-var ErrConflictingRefs = errors.New("conflicting references")
+var (
+	ErrConflictingRefs = errors.New("conflicting references")
+	ErrWrongStatus     = errors.New("incorrect status")
+)
 
 // ExportBranchDone ends the export branch process by changing the status
-func ExportBranchDone(cataloger catalog.Cataloger, status catalog.CatalogBranchExportStatus, statusMsg *string, repo, branch, commitRef string) error {
-	err := cataloger.ExportStateSet(repo, branch, func(oldRef string, state catalog.CatalogBranchExportStatus) (newRef string, newState catalog.CatalogBranchExportStatus, newMessage *string, err error) {
+func ExportBranchDone(parade parade.Parade, cataloger catalog.Cataloger, status catalog.CatalogBranchExportStatus, statusMsg *string, repo, branch, commitRef string) error {
+	l := logging.Default().WithFields(logging.Fields{"repo": repo, "branch": branch, "commit_ref": commitRef, "status": status, "status_message": statusMsg})
+
+	err := cataloger.ExportStateSet(repo, branch, func(oldRef string, oldStatus catalog.CatalogBranchExportStatus) (newRef string, newStatus catalog.CatalogBranchExportStatus, newMessage *string, err error) {
 		if commitRef != oldRef {
-			return "", "", nil, fmt.Errorf("ExportBranchDone: currentRef:%s, newRef:%s: %w", oldRef, commitRef, ErrConflictingRefs)
+			return "", "", nil, fmt.Errorf("ExportBranchDone: currentRef: %s, newRef: %s: %w", oldRef, commitRef, ErrConflictingRefs)
+		}
+		if oldStatus != catalog.ExportStatusInProgress {
+			l.WithField("old_state", oldStatus).Error("expected old_state to be in-progress")
+			return "", "", nil, fmt.Errorf("expected old status to be in-progress not %s: %w", oldStatus, ErrWrongStatus)
 		}
+		l.WithField("old_state", oldStatus).Info("updating branch export")
 		return oldRef, status, statusMsg, nil
 	})
-	return err
-}
+	if err != nil {
+		return err
+	}
 
-var ErrRepairWrongStatus = errors.New("incorrect status")
+	if status == catalog.ExportStatusSuccess {
+		// Start the next export if continuous.
+		isContinuous, err := hasContinuousExport(cataloger, repo, branch)
+		if err != nil {
+			// Consider branch export failed: it was supposed to be continuous but
+			// might have stopped.  So set an error for the admin to fix before
+			// re-enabling continuous export.
+			return err
+		}
+		if isContinuous {
+			l.Info("start new continuous export")
+			_, err := ExportBranchStart(parade, cataloger, repo, branch)
+			if errors.Is(err, ErrExportInProgress) {
+				l.Info("export already in progress when restarting continuous export (unlikely)")
+				err = nil
+			} else if errors.Is(err, ErrNothingToExport) {
+				l.Info("nothing further to export continuously; stop (likely)")
+				err = nil
+			}
+			if err != nil {
+				return fmt.Errorf("restart continuous export repo %s branch %s: %w", repo, branch, err)
+			}
+			return nil
```

Comment:
```suggestion
			if errors.Is(err, ErrExportInProgress) {
				l.Info("export already in progress when restarting continuous export (unlikely)")
				return nil
			}
			if errors.Is(err, ErrNothingToExport) {
				l.Info("nothing further to export continuously; stop (likely)")
				return nil
			}
			if err != nil {
				return fmt.Errorf("restart continuous export repo %s branch %s: %w", repo, branch, err)
			}
			return nil
```

Reply:
Not done.  I prefer to have a single "good return" for the function, it's a place to add stuff that has to happen after.  Early returns are good for errors, less for mixed errors and successes.